### DISCUSSION TITLE
实验代码中，ID模块有无用的指令集

### DISCUSSION THREAD

**Discussion by WillanSnow** [SYSU-SCC/yatcpu-docs#17] [General]
为了方便查看波形，我尝试总结实验代码中出现的指令集合。
我发现op段有一个**RM**型指令，且R、M型指令内部的funct7编码类似。
![image](https://user-images.githubusercontent.com/105115785/213394179-8ceaa96f-5334-4f72-83be-83a93c42e7af.png)![image](https://user-images.githubusercontent.com/105115785/213394207-59de9d8c-79f6-494e-a14c-5b165efdecc6.png)

我很好奇他们的区分方法，就去ALU内部查看细节。发现ALU内部并没有M型指令的实现，乘除之类的。
![image](https://user-images.githubusercontent.com/105115785/213394664-5f8b36c8-9413-4cb0-a0d7-84d5a0b00fe5.png)

我继续编写了有乘法运算的c代码，并尝试编译成.asmbin文件，看看cpu如何实现。
![image](https://user-images.githubusercontent.com/105115785/213395039-6d38481f-55ea-45fe-8091-bc5663cf1151.png)
结果在运行build.bat时，报错，无法执行。就是说**CPU不支持乘法指令**（可能也不支持所有的M型指令）
![image](https://user-images.githubusercontent.com/105115785/213395488-d66d6f3c-455a-481d-8265-713fd1727e49.png)
*将a*b修改成a+b后可正常执行*
*又偷偷跑去看了yatcpu内部的代码，里面的ALU好像也没有M型指令的支持*

**Comment by howardlau1999** [comment]
RM型指令那里是opcode，第二张图是funct3，是加法还是乘法用funct7判断，RV32I没有要求实现乘除法

**Reply by WillanSnow** [reply]
懂了，谢谢助教。
我理解错了，我想着0-7刚好占满，RM型指令op一致要做区分就应该不够了，误以为他们是funct7了。
